src/Microsoft.Sbom.Api/Workflows/Helpers/RelationshipsArrayGenerator.cs

```diff
@@ -26,92 +26,92 @@ public class RelationshipsArrayGenerator : IJsonArrayGenerator<RelationshipsArra
 
     private readonly ILogger log;
 
-    private readonly ISbomConfigProvider sbomConfigs;
-
     private readonly IRecorder recorder;
 
+    public ISbomConfig SbomConfig { get; set; }
+
+    public string SpdxManifestVersion { get; set; }
+
     public RelationshipsArrayGenerator(
         RelationshipGenerator generator,
         ChannelUtils channelUtils,
         ILogger log,
-        ISbomConfigProvider sbomConfigs,
         IRecorder recorder)
     {
         this.generator = generator;
         this.channelUtils = channelUtils;
         this.log = log;
-        this.sbomConfigs = sbomConfigs;
         this.recorder = recorder;
     }
 
-    public async Task<IList<FileValidationResult>> GenerateAsync()
+    public async Task<GenerateResult> GenerateAsync()
     {
         using (recorder.TraceEvent(Events.RelationshipsGeneration))
         {
-            IList<FileValidationResult> totalErrors = new List<FileValidationResult>();
+            var totalErrors = new List<FileValidationResult>();
+            var serializersToJsonDocs = new Dictionary<IManifestToolJsonSerializer, List<JsonDocument>>();
+
+            IList<ISbomConfig> relationshipsArraySupportingConfigs = new List<ISbomConfig>();
+            var serializationStrategy = JsonSerializationStrategyFactory.GetStrategy(SpdxManifestVersion);
 
             // Write the relationship array only if supported
-            foreach (var manifestInfo in sbomConfigs.GetManifestInfos())
+            if (serializationStrategy.AddToRelationshipsSupportingConfig(ref relationshipsArraySupportingConfigs, this.SbomConfig))
             {
-                var sbomConfig = sbomConfigs.Get(manifestInfo);
-                if (sbomConfig.MetadataBuilder.TryGetRelationshipsHeaderName(out var relationshipArrayHeaderName))
+                // Get generation data
+                var generationData = this.SbomConfig.Recorder.GetGenerationData();
```

Comment:
I'm strongly biased against adding `this.` outside of a constructor unless it's absolutely necessary. I would love to see us avoid adding new ones if we can

Reply:
I moved this logic to the constructor, does that achieve what you had in mind? The reason I'm having to do this instead of the previous code is because I moved the uber for loop that all these generators had to the caller instead of having them in individual generators